compute_sdk/globus_compute_sdk/sdk/client.py

```diff
@@ -120,6 +114,11 @@ def __init__(
             )
             warnings.warn(msg)
 
+        if funcx_service_address is None:
+            funcx_service_address = get_web_service_url(environment)
+
+        self._task_status_table: t.Dict[str, t.Dict] = {}
+
```

Comment:
Since you're touching this line anyway, consider upgrading the typing Py310 style:
```python
self._task_status_table: dict[str, dict] = {}
```